fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -51,7 +51,7 @@ def __init__(
         world_size: int,
         broadcast_buffers: bool,
         process_group: Any = None,
-        buffer_size: int = 2 ** 28,
+        buffer_size: int = 2 ** 20,
```

Comment:
You may want to consider having two parameters. A max size for the buffer and a min size below which a parameter is bucketed. Those feel like two separate concerns. Otherwise, you end up bucketing parameters that are very close to buffer_size. That would be a waste.

Reply:
it's not the same mechanism as before, the params are sorted by size and only the smallest ones are bucketed, and this happens only once so that the whole thing can stays async. This results in a huge perf win for shardedDDP

Reply:
the worst case is if the smallest parameter is just below the buffer size, this means *one* extra copy for no gain. All other cases that I can think of are what looks like reasonable tradeoffs. The benchmark is with a ResNet101, hopefully close to something typical for vision